src/main/java/io/quarkus/gizmo/BytecodeCreatorImpl.java

```diff
@@ -342,7 +354,16 @@ public ResultHandle loadClass(String className) {
             primitiveType = Void.class;
         }
         if (primitiveType == null) {
-            return new ResultHandle("Ljava/lang/Class;", this, Type.getObjectType(className.replace('.', '/')));
+            if (useTccl) {
+                if (cachedTccl == null) {
+                    ResultHandle currentThread = invokeStaticMethod(THREAD_CURRENT_THREAD);
+                   cachedTccl = invokeVirtualMethod(THREAD_GET_TCCL, currentThread);
+                }
+                return invokeStaticMethod(CL_FOR_NAME,
+                                load(className), load(false), cachedTccl);
```

Comment:
Shouldn't we allow users to speficy the `initialize` value?

Reply:
Not really, its kinda pointless IMHO unless you really want to force static init to run. If they really want to do that they can call the class manually.

Reply:
My point is that `java.lang.Class.forName(String)` has `initialize=true` so it's IMO kind of expected to force static init.